File: php/src/vaas/Http/Options/ForStreamOptions.php
<?php

namespace VaasSdk\Options;

class ForStreamOptions
{
    public bool $useHashLookup;
    public ?string $vaasRequestId;

    public function __construct(array $options = [])
    {
        $this->useHashLookup = $options['useHashLookup'] ?? true;
        $this->vaasRequestId = $options['vaasRequestId'] ?? null;
    }

    public static function default(): self
    {
        return new self();
    }
}

File: java/src/main/java/de/gdata/vaas/options/ForStreamOptions.java
package de.gdata.vaas.options;

import de.gdata.vaas.VaasConfig;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.jetbrains.annotations.Nullable;

/**
 * Options for configuring forStream requests.
 * It includes options for using hash lookup and specifying a Vaas request ID.
 *
 * <p>
 * The {@code UseHashLookup} field indicates whether hash lookup should be used.
 * The {@code VaasRequestId} field can hold a nullable string representing the Vaas request ID.
 * </p>
 */
@Setter
@Getter
@AllArgsConstructor
@NoArgsConstructor
public class ForStreamOptions {
    boolean UseHashLookup = true;

    @Nullable
    String VaasRequestId;

    public static ForStreamOptions fromVaasConfig(VaasConfig config) {
        return new ForStreamOptions(config.isUseHashLookup(), null);
    }
}
